File: carne_generator.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Gerador de Carnês de Pagamento em PDF
Especificações: 4 cupons por página A4, 2 vias (cobrador e cliente)
"""

from reportlab.lib.pagesizes import A4
from reportlab.lib.units import mm
from reportlab.lib.colors import HexColor, Color, black, white
from reportlab.pdfgen import canvas
from reportlab.lib.styles import ParagraphStyle, getSampleStyleSheet
from reportlab.platypus import SimpleDocTemplate, PageBreak, Spacer, Table, TableStyle, Paragraph
from datetime import datetime, timedelta
from reportlab.lib import colors
import os

# ───────────────────────────────────────────────────────────────────────────────
# CONFIGURAÇÃO DE DADOS
# ───────────────────────────────────────────────────────────────────────────────

EMPRESA = "Posthumous"
SUBTITULO = "Gestão de Serviços Póstumos"
CLIENTE = "Ana Costa"
CPF = "123.456.789-00"
TOTAL = 3050.00
PARCELAS = 12
CARNE_ID = "2026-4567"

# Paleta de cores - Azul Corporativo (como no HTML)
PALETA = {
    "dark": HexColor("#1e3a5f"),
    "accent": HexColor("#2563eb"),
    "light": HexColor("#dbeafe"),
    "xlight": HexColor("#eff6ff"),
    "text": HexColor("#1e293b"),
    "label": HexColor("#93c5fd"),
    "border": HexColor("#bfdbfe"),
    "perf": HexColor("#60a5fa"),
}

# Cálculos
VALOR_PARCELA = TOTAL / PARCELAS
DATA_BASE = datetime(2026, 9, 1)

# ───────────────────────────────────────────────────────────────────────────────
# GERAÇÃO DO PDF
# ───────────────────────────────────────────────────────────────────────────────

class CarneGenerator:
    def __init__(self, filename="carne_pagamentos.pdf"):
        self.filename = filename
        self.page_width, self.page_height = A4
        self.margin = 10 * mm
        self.cupom_height = 65.5 * mm
        self.cupom_width = self.page_width - (2 * self.margin)
        
    def draw_cupom(self, c, y_pos, numero_parcela):
        """Desenha um cupom com 2 vias"""
        x_start = self.margin
        cupom_width = self.cupom_width
        cupom_height = self.cupom_height
        
        # Calcula datas
        mes = (numero_parcela - 1) % 12
        ano_ref = 2026 + ((numero_parcela - 1) // 12)
        data_ref = f"{numero_parcela:02d}/2026"
        data_venc = (DATA_BASE + timedelta(days=30 * (numero_parcela - 1))).strftime("%d/%m/%Y")
        
        # ────────────────────────────────────────────────────────────────────
        # VIA COBRADOR (esquerda - 38%)
        # ────────────────────────────────────────────────────────────────────
        via_cobrador_width = cupom_width * 0.38
        
        # Fundo claro
        c.setFillColor(PALETA["light"])
        c.rect(x_start, y_pos, via_cobrador_width, cupom_height, fill=1, stroke=0)
        
        # Barra superior escura
        barra_height = 7 * mm
        c.setFillColor(PALETA["dark"])
        c.rect(x_start, y_pos + cupom_height - barra_height, via_cobrador_width, barra_height, fill=1, stroke=0)
        
        # Texto barra: empresa e via
        c.setFont("Helvetica-Bold", 8)
        c.setFillColor(white)
        c.drawString(x_start + 3 * mm, y_pos + cupom_height - barra_height + 2 * mm, EMPRESA)
        c.drawRightString(x_start + via_cobrador_width - 2 * mm, y_pos + cupom_height - barra_height + 2 * mm, "VIA COBRADOR")
        
        # Conteúdo via cobrador
        c.setFont("Helvetica", 6.5)
        c.setFillColor(PALETA["text"])
        
        y_content = y_pos + cupom_height - barra_height - 4 * mm
        
        # Cliente
        c.setFillColor(PALETA["accent"])
        c.setFont("Helvetica-Bold", 5.5)
        c.drawString(x_start + 3 * mm, y_content, "CLIENTE")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(x_start + 3 * mm, y_content - 2.5 * mm, CLIENTE)
        
        # Referência e Vencimento
        y_content -= 7 * mm
        c.setFillColor(PALETA["accent"])
        c.setFont("Helvetica-Bold", 5.5)
        c.drawString(x_start + 3 * mm, y_content, "REF.")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(x_start + 3 * mm, y_content - 2.5 * mm, data_ref)
        
        c.setFillColor(PALETA["accent"])
        c.setFont("Helvetica-Bold", 5.5)
        c.drawString(x_start + 3 * mm, y_content - 5 * mm, "VENC.")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(x_start + 3 * mm, y_content - 7.5 * mm, data_venc)
        
        # Número da parcela (grande)
        c.setFillColor(PALETA["accent"])
        c.setFont("Helvetica-Bold", 14)
        parcela_text = f"{numero_parcela:02d}/{PARCELAS:02d}"
        c.drawRightString(x_start + via_cobrador_width - 3 * mm, y_pos + 3 * mm, parcela_text)
        
        # ────────────────────────────────────────────────────────────────────
        # LINHA PERFURADA (vertical)
        # ────────────────────────────────────────────────────────────────────
        perf_x = x_start + via_cobrador_width + 2.5 * mm
        c.setStrokeColor(PALETA["perf"])
        c.setLineWidth(0.5)
        dash_height = 2 * mm
        dash_gap = 1.5 * mm
        
        for y in range(int(y_pos), int(y_pos + cupom_height), int(dash_height + dash_gap)):
            c.line(perf_x, y, perf_x, min(y + dash_height, y_pos + cupom_height))
        
        # ────────────────────────────────────────────────────────────────────
        # VIA CLIENTE (direita - 62%)
        # ────────────────────────────────────────────────────────────────────
        via_cliente_x = x_start + via_cobrador_width + 5 * mm
        via_cliente_width = cupom_width - via_cobrador_width - 5 * mm
        
        # Fundo branco
        c.setFillColor(white)
        c.rect(via_cliente_x, y_pos, via_cliente_width, cupom_height, fill=1, stroke=1)
        c.setStrokeColor(PALETA["border"])
        c.setLineWidth(1)
        c.rect(via_cliente_x, y_pos, via_cliente_width, cupom_height, stroke=1)
        
        # Barra superior
        c.setFillColor(PALETA["dark"])
        c.rect(via_cliente_x, y_pos + cupom_height - barra_height, via_cliente_width, barra_height, fill=1, stroke=0)
        
        c.setFont("Helvetica-Bold", 8)
        c.setFillColor(white)
        c.drawString(via_cliente_x + 3 * mm, y_pos + cupom_height - barra_height + 2 * mm, EMPRESA)
        c.drawRightString(via_cliente_x + via_cliente_width - 2 * mm, y_pos + cupom_height - barra_height + 2 * mm, "VIA CLIENTE/RECIBO")
        
        # Conteúdo dividido em 2 colunas
        col_width = (via_cliente_width - 6 * mm) / 2
        col1_x = via_cliente_x + 3 * mm
        col2_x = via_cliente_x + col_width + 3 * mm
        
        y_content = y_pos + cupom_height - barra_height - 3 * mm
        
        # Coluna 1
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#64748b"))
        c.drawString(col1_x, y_content, "CLIENTE")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(col1_x, y_content - 2.5 * mm, CLIENTE)
        
        y_content -= 6 * mm
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#64748b"))
        c.drawString(col1_x, y_content, "REFERÊNCIA")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(col1_x, y_content - 2.5 * mm, data_ref)
        
        y_content -= 6 * mm
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#64748b"))
        c.drawString(col1_x, y_content, "VENCIMENTO")
        c.setFont("Helvetica", 6)
        c.setFillColor(PALETA["text"])
        c.drawString(col1_x, y_content - 2.5 * mm, data_venc)
        
        # Coluna 2
        y_content = y_pos + cupom_height - barra_height - 3 * mm
        
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#64748b"))
        c.drawString(col2_x, y_content, "PARCELA")
        c.setFont("Helvetica-Bold", 11)
        c.setFillColor(PALETA["accent"])
        c.drawString(col2_x, y_content - 3.5 * mm, parcela_text)
        
        y_content -= 9 * mm
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#64748b"))
        c.drawString(col2_x, y_content, "VALOR")
        c.setFont("Helvetica-Bold", 9)
        c.setFillColor(PALETA["dark"])
        valor_text = f"R$ {VALOR_PARCELA:,.2f}".replace(".", ",")
        c.drawString(col2_x, y_content - 2.5 * mm, valor_text)
        
        # Badge de status
        y_content -= 6 * mm
        c.setFillColor(HexColor("#fef3c7"))
        c.rect(col2_x, y_content - 2.5 * mm, 16 * mm, 3 * mm, fill=1, stroke=0)
        c.setFont("Helvetica-Bold", 5)
        c.setFillColor(HexColor("#b45309"))
        c.drawString(col2_x + 1.5 * mm, y_content - 1.5 * mm, "PENDENTE")
        
    def draw_linha_corte(self, c, y_pos):
        """Desenha linha de corte entre cupons"""
        c.setStrokeColor(HexColor("#cbd5e1"))
        c.setLineWidth(0.5)
        dash = [2, 2]
        c.setDash(dash)
        
        x_start = self.margin
        x_end = self.page_width - self.margin
        
        # Símbolo de tesoura
        c.setDash([])
        c.setFont("Helvetica", 8)
        c.setFillColor(HexColor("#cbd5e1"))
        c.drawString((self.page_width - 4 * mm) / 2, y_pos - 1.5 * mm, "✂")
        
        # Linhas
        c.setDash(dash)
        c.line(x_start, y_pos, (self.page_width - 6 * mm) / 2, y_pos)
        c.line((self.page_width + 6 * mm) / 2, y_pos, x_end, y_pos)
        c.setDash([])
        
    def generate(self):
        """Gera o PDF com 4 cupons por página"""
        # Cria PDF
        c = canvas.Canvas(self.filename, pagesize=A4)
        
        # Adiciona metadados
        c.setTitle("Carnê de Pagamento - Posthumous")
        c.setAuthor("Posthumous - Sistema de Gestão")
        
        cupom_atual = 1
        y_pos = self.page_height - self.margin - self.cupom_height
        cupons_por_pagina = 4
        
        # Total de parcelas (por padrão 12)
        for i in range(PARCELAS):
            # Desenha cupom
            self.draw_cupom(c, y_pos, cupom_atual)
            
            # Próxima posição
            y_pos -= self.cupom_height + 2 * mm
            
            # Linha de corte
            if i < PARCELAS - 1:
                self.draw_linha_corte(c, y_pos)
                y_pos -= 3 * mm
            
            cupom_atual += 1
            
            # Pula página após 4 cupons
            if cupom_atual % (cupons_por_pagina + 1) == 1 and i < PARCELAS - 1:
                c.showPage()
                y_pos = self.page_height - self.margin - self.cupom_height
        
        # Rodapé informações
        c.setFont("Helvetica", 8)
        c.setFillColor(HexColor("#6b7280"))
        c.drawString(self.margin, 8 * mm, f"Carnê nº {CARNE_ID} • Cliente: {CLIENTE} • CPF: {CPF}")
        c.drawString(self.margin, 5 * mm, f"Total: R$ {TOTAL:,.2f} • Parcelas: {PARCELAS}x R$ {VALOR_PARCELA:,.2f}")
        c.drawString(self.margin, 2 * mm, f"Gerado em: {datetime.now().strftime('%d/%m/%Y às %H:%M')}")
        
        # Salva
        c.save()
        print(f"✓ PDF gerado com sucesso: {self.filename}")
        print(f"  - {PARCELAS} parcelas")
        print(f"  - Total: R$ {TOTAL:,.2f}")
        print(f"  - Valor parcela: R$ {VALOR_PARCELA:,.2f}")

# ───────────────────────────────────────────────────────────────────────────────
# EXECUÇÃO
# ───────────────────────────────────────────────────────────────────────────────

if __name__ == "__main__":
    output_path = os.path.expanduser("~/Downloads/carne_pagamentos.pdf")
    generator = CarneGenerator(output_path)
    generator.generate()

File: carne_generator.js
#!/usr/bin/env node
/**
 * Gerador de Carnês de Pagamento em PDF
 * Especificações: 4 cupons por página A4, 2 vias (cobrador e cliente)
 * 
 * Instalar dependências: npm install pdfkit
 */

const PDFDocument = require('pdfkit');
const fs = require('fs');
const path = require('path');

// ───────────────────────────────────────────────────────────────────────────────
// CONFIGURAÇÃO DE DADOS
// ───────────────────────────────────────────────────────────────────────────────

const CONFIG = {
  empresa: "Posthumous",
  subtitulo: "Gestão de Serviços Póstumos",
  cliente: "Ana Costa",
  cpf: "123.456.789-00",
  total: 3050.00,
  parcelas: 12,
  carneId: "2026-4567",
};

// Paleta de cores - Azul Corporativo (conforme HTML)
const PALETA = {
  dark: "#1e3a5f",
  accent: "#2563eb",
  light: "#dbeafe",
  xlight: "#eff6ff",
  text: "#1e293b",
  label: "#93c5fd",
  border: "#bfdbfe",
  perf: "#60a5fa",
};

// ───────────────────────────────────────────────────────────────────────────────
// CLASSE GERADOR
// ───────────────────────────────────────────────────────────────────────────────

class CarneGenerator {
  constructor(outputPath) {
    this.outputPath = outputPath;
    this.doc = new PDFDocument({
      size: 'A4',
      margin: 10,
      bufferPages: true,
    });
    this.pageWidth = this.doc.page.width;
    this.pageHeight = this.doc.page.height;
    this.margin = 10;
    this.cupomHeight = 185.5; // ~65.5mm em pt
    this.cupomWidth = this.pageWidth - (2 * this.margin);
    this.valorParcela = CONFIG.total / CONFIG.parcelas;
    this.dataBase = new Date(2026, 8, 1); // Setembro (mês 8)
  }

  /**
   * Formata número para formato monetário brasileiro
   */
  formatMoney(value) {
    return new Intl.NumberFormat('pt-BR', {
      style: 'currency',
      currency: 'BRL',
    }).format(value);
  }

  /**
   * Calcula data de vencimento para a parcela
   */
  getDataVencimento(numeroParcela) {
    const data = new Date(this.dataBase);
    data.setMonth(data.getMonth() + (numeroParcela - 1));
    return data.toLocaleDateString('pt-BR');
  }

  /**
   * Desenha um cupom com 2 vias
   */
  drawCupom(yPos, numeroParcela) {
    const xStart = this.margin;
    const cupomWidth = this.cupomWidth;
    const cupomHeight = this.cupomHeight;

    // Dados da parcela
    const dataRef = `${String(numeroParcela).padStart(2, '0')}/2026`;
    const dataVenc = this.getDataVencimento(numeroParcela);

    // ────────────────────────────────────────────────────────────────────
    // VIA COBRADOR (esquerda - 38%)
    // ────────────────────────────────────────────────────────────────────
    const viaCobradorWidth = cupomWidth * 0.38;

    // Fundo claro
    this.doc.rect(xStart, yPos, viaCobradorWidth, cupomHeight)
      .fillAndStroke(PALETA.light, PALETA.border);

    // Barra superior escura
    const barraHeight = 20;
    this.doc.rect(xStart, yPos + cupomHeight - barraHeight, viaCobradorWidth, barraHeight)
      .fill(PALETA.dark);

    // Texto barra
    this.doc.fontSize(7).fillColor('white').font('Helvetica-Bold');
    this.doc.text(CONFIG.empresa, xStart + 8, yPos + cupomHeight - barraHeight + 4, {
      width: viaCobradorWidth - 16,
      align: 'left',
    });

    this.doc.fontSize(6).text('VIA COBRADOR', xStart + 8, yPos + cupomHeight - barraHeight + 10, {
      width: viaCobradorWidth - 16,
      align: 'right',
    });

    // Conteúdo via cobrador
    let yContent = yPos + cupomHeight - barraHeight - 10;

    this.doc.fontSize(5).fillColor(PALETA.accent).font('Helvetica-Bold');
    this.doc.text('CLIENTE', xStart + 8, yContent);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(CONFIG.cliente, xStart + 8, yContent + 8);

    yContent -= 25;

    this.doc.fontSize(5).fillColor(PALETA.accent).font('Helvetica-Bold');
    this.doc.text('REF.', xStart + 8, yContent);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(dataRef, xStart + 8, yContent + 8);

    this.doc.fontSize(5).fillColor(PALETA.accent).font('Helvetica-Bold');
    this.doc.text('VENC.', xStart + 8, yContent + 20);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(dataVenc, xStart + 8, yContent + 28);

    // Número da parcela (grande)
    this.doc.fontSize(16).fillColor(PALETA.accent).font('Helvetica-Bold');
    const parcelaText = `${String(numeroParcela).padStart(2, '0')}/${String(CONFIG.parcelas).padStart(2, '0')}`;
    this.doc.text(
      parcelaText,
      xStart + 8,
      yPos + 10,
      { width: viaCobradorWidth - 16, align: 'right' }
    );

    // ────────────────────────────────────────────────────────────────────
    // LINHA PERFURADA (vertical)
    // ────────────────────────────────────────────────────────────────────
    const perfX = xStart + viaCobradorWidth + 7;
    this.doc.strokeColor(PALETA.perf).lineWidth(0.5);

    const dashHeight = 5;
    const dashGap = 4;
    for (let y = yPos; y < yPos + cupomHeight; y += dashHeight + dashGap) {
      this.doc.moveTo(perfX, y).lineTo(perfX, Math.min(y + dashHeight, yPos + cupomHeight)).stroke();
    }

    // ────────────────────────────────────────────────────────────────────
    // VIA CLIENTE (direita - 62%)
    // ────────────────────────────────────────────────────────────────────
    const viaClienteX = xStart + viaCobradorWidth + 15;
    const viaClienteWidth = cupomWidth - viaCobradorWidth - 15;

    // Fundo e borda
    this.doc.rect(viaClienteX, yPos, viaClienteWidth, cupomHeight)
      .fillAndStroke('white', PALETA.border);

    // Barra superior
    this.doc.rect(viaClienteX, yPos + cupomHeight - barraHeight, viaClienteWidth, barraHeight)
      .fill(PALETA.dark);

    // Texto barra
    this.doc.fontSize(7).fillColor('white').font('Helvetica-Bold');
    this.doc.text(CONFIG.empresa, viaClienteX + 8, yPos + cupomHeight - barraHeight + 4, {
      width: viaClienteWidth - 16,
      align: 'left',
    });

    this.doc.fontSize(6).text('VIA CLIENTE/RECIBO', viaClienteX + 8, yPos + cupomHeight - barraHeight + 10, {
      width: viaClienteWidth - 16,
      align: 'right',
    });

    // Conteúdo em 2 colunas
    const colWidth = (viaClienteWidth - 14) / 2;
    const col1X = viaClienteX + 8;
    const col2X = viaClienteX + colWidth + 8;

    yContent = yPos + cupomHeight - barraHeight - 10;

    // Coluna 1
    this.doc.fontSize(5).fillColor('#64748b').font('Helvetica-Bold');
    this.doc.text('CLIENTE', col1X, yContent);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(CONFIG.cliente, col1X, yContent + 8, { width: colWidth });

    yContent -= 25;

    this.doc.fontSize(5).fillColor('#64748b').font('Helvetica-Bold');
    this.doc.text('REFERÊNCIA', col1X, yContent);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(dataRef, col1X, yContent + 8, { width: colWidth });

    yContent -= 25;

    this.doc.fontSize(5).fillColor('#64748b').font('Helvetica-Bold');
    this.doc.text('VENCIMENTO', col1X, yContent);

    this.doc.fontSize(6).fillColor(PALETA.text).font('Helvetica');
    this.doc.text(dataVenc, col1X, yContent + 8, { width: colWidth });

    // Coluna 2
    yContent = yPos + cupomHeight - barraHeight - 10;

    this.doc.fontSize(5).fillColor('#64748b').font('Helvetica-Bold');
    this.doc.text('PARCELA', col2X, yContent);

    this.doc.fontSize(12).fillColor(PALETA.accent).font('Helvetica-Bold');
    this.doc.text(parcelaText, col2X, yContent + 8, { width: colWidth, align: 'left' });

    yContent -= 25;

    this.doc.fontSize(5).fillColor('#64748b').font('Helvetica-Bold');
    this.doc.text('VALOR', col2X, yContent);

    this.doc.fontSize(10).fillColor(PALETA.dark).font('Helvetica-Bold');
    this.doc.text(this.formatMoney(this.valorParcela), col2X, yContent + 8, {
      width: colWidth,
      align: 'left',
    });

    // Badge de status
    yContent -= 20;
    this.doc.rect(col2X, yContent - 8, 45, 10).fill('#fef3c7');

    this.doc.fontSize(5).fillColor('#b45309').font('Helvetica-Bold');
    this.doc.text('PENDENTE', col2X + 2, yContent - 6, { width: 41, align: 'center' });
  }

  /**
   * Desenha linha de corte entre cupons
   */
  drawLinhaCortе(yPos) {
    this.doc.strokeColor('#cbd5e1').lineWidth(0.5);
    this.doc.dash(2, { space: 2 });

    const xStart = this.margin;
    const xEnd = this.pageWidth - this.margin;
    const midX = this.pageWidth / 2;

    // Linhas perfuradas
    this.doc.moveTo(xStart, yPos).lineTo(midX - 15, yPos).stroke();
    this.doc.moveTo(midX + 15, yPos).lineTo(xEnd, yPos).stroke();

    // Ícone de tesoura
    this.doc.undash();
    this.doc.fontSize(10).fillColor('#cbd5e1').font('Helvetica');
    this.doc.text('✂', midX - 8, yPos - 8, { width: 16, align: 'center' });
  }

  /**
   * Gera o PDF
   */
  generate() {
    console.log('📄 Gerando carnê de pagamento...\n');

    let cupomAtual = 1;
    let yPos = this.pageHeight - this.margin - this.cupomHeight;
    const cuponsPerPage = 4;

    // Itera por todas as parcelas
    for (let i = 0; i < CONFIG.parcelas; i++) {
      // Desenha cupom
      this.drawCupom(yPos, cupomAtual);

      // Próxima posição
      yPos -= this.cupomHeight + 6;

      // Linha de corte
      if (i < CONFIG.parcelas - 1) {
        this.drawLinhaCortе(yPos);
        yPos -= 8;
      }

      cupomAtual++;

      // Pula página após 4 cupons
      if (cupomAtual % (cuponsPerPage + 1) === 1 && i < CONFIG.parcelas - 1) {
        this.doc.addPage();
        yPos = this.pageHeight - this.margin - this.cupomHeight;
      }
    }

    // Rodapé com informações
    this.doc.fontSize(8).fillColor('#6b7280').font('Helvetica');
    this.doc.text(
      `Carnê nº ${CONFIG.carneId} • Cliente: ${CONFIG.cliente} • CPF: ${CONFIG.cpf}`,
      this.margin,
      this.pageHeight - 25
    );

    this.doc.text(
      `Total: ${this.formatMoney(CONFIG.total)} • Parcelas: ${CONFIG.parcelas}x ${this.formatMoney(this.valorParcela)}`,
      this.margin,
      this.pageHeight - 15
    );

    const now = new Date();
    const dataHora = now.toLocaleDateString('pt-BR') + ' às ' + now.toLocaleTimeString('pt-BR');
    this.doc.text(`Gerado em: ${dataHora}`, this.margin, this.pageHeight - 5);

    // Salva arquivo
    const stream = fs.createWriteStream(this.outputPath);
    this.doc.pipe(stream);
    this.doc.end();

    return new Promise((resolve, reject) => {
      stream.on('finish', () => {
        console.log('✓ PDF gerado com sucesso!\n');
        console.log(`📁 Arquivo: ${this.outputPath}`);
        console.log(`📊 Detalhes:`);
        console.log(`   - ${CONFIG.parcelas} parcelas`);
        console.log(`   - Total: ${this.formatMoney(CONFIG.total)}`);
        console.log(`   - Valor parcela: ${this.formatMoney(this.valorParcela)}`);
        console.log(`   - Cliente: ${CONFIG.cliente}`);
        console.log(`   - CPF: ${CONFIG.cpf}`);
        console.log(`   - Carnê nº: ${CONFIG.carneId}\n`);
        resolve();
      });

      stream.on('error', reject);
      this.doc.on('error', reject);
    });
  }
}

// ───────────────────────────────────────────────────────────────────────────────
// EXECUÇÃO
// ───────────────────────────────────────────────────────────────────────────────

async function main() {
  try {
    const outputPath = path.join(
      require('os').homedir(),
      'Downloads',
      'carne_pagamentos.pdf'
    );

    const generator = new CarneGenerator(outputPath);
    await generator.generate();
  } catch (error) {
    console.error('❌ Erro ao gerar PDF:', error.message);
    process.exit(1);
  }
}

main();
